webui/react/src/components/FilterForm/components/FilterField.tsx

```diff
@@ -167,86 +176,91 @@ const FilterField = ({
         }}
       />
       <div className={css.fieldCard} ref={preview}>
-        <>
-          <Select
-            autoFocus
-            dropdownMatchSelectWidth={250}
-            options={columns.map((col) => ({
-              label: col.displayName || col.column,
-              value: col.column,
-            }))}
-            value={field.columnName}
-            width={'100%'}
-            onChange={onChangeColumnName}
-          />
-          <Select
-            options={(isSpecialColumn
-              ? [Operator.Eq, Operator.NotEq] // just Eq and NotEq for Special column
-              : AvailableOperators[currentColumn?.type ?? V1ColumnType.UNSPECIFIED]
-            ).map((op) => ({
-              label: ReadableOperator[field.type][op],
-              value: op,
-            }))}
-            value={field.operator}
-            width={'100%'}
-            onChange={(value) => {
-              const op = (value?.toString() ?? '=') as Operator;
-              formStore.setFieldOperator(field.id, op);
-              if (op === Operator.IsEmpty || op === Operator.NotEmpty) {
-                updateFieldValue(field.id, null);
-              }
-            }}
-          />
-          {isSpecialColumn ? (
-            <>
-              <Select
-                options={getSpecialOptions(field.columnName as SpecialColumnNames)}
-                value={fieldValue?.toString()}
-                width={'100%'}
-                onChange={(value) => {
-                  const val = value?.toString() ?? null;
-                  updateFieldValue(field.id, val);
+        <Select
+          autoFocus
+          dropdownMatchSelectWidth={250}
+          options={columns.map((col) => ({
+            label: col.displayName || col.column,
+            value: col.column,
+          }))}
+          value={field.columnName}
+          width={'100%'}
+          onChange={onChangeColumnName}
+        />
+        <Select
+          options={(isSpecialColumn
+            ? [Operator.Eq, Operator.NotEq] // just Eq and NotEq for Special column
+            : AvailableOperators[currentColumn?.type ?? V1ColumnType.UNSPECIFIED]
+          ).map((op) => ({
+            label: ReadableOperator[field.type][op],
+            value: op,
+          }))}
+          value={field.operator}
+          width={'100%'}
+          onChange={(value) => {
+            const op = (value?.toString() ?? '=') as Operator;
+            formStore.setFieldOperator(field.id, op);
+            if (op === Operator.IsEmpty || op === Operator.NotEmpty) {
+              updateFieldValue(field.id, null);
+            }
+          }}
+        />
+        {isSpecialColumn ? (
+          <div onKeyDownCapture={captureEnterKeyDown}>
```

Comment:
this behavior is a bit weird to me. It selects an option but every time user press enter, it triggers to add a new field. Do you know if we want this feature for non-text input fields like select, date inputs?

https://github.com/determined-ai/determined/assets/109113616/6e3e2a02-c881-4972-8e84-c60b205e37a5

Reply:
@alexjdetermined thoughts? I'm of the opinion that it makes sense for the enter key to work the same across the board, so pressing enter on the value field should create a new field. In addition, because we only send valid sorts to the search, it shouldn't interfere with the user when this happens.

Reply:
In this case you should have to hit "Enter" twice in order to get the new row, once to select an item and once to advance to the new row.